scripts/validate/govcms-validate-modules

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+# shellcheck disable=SC2162,SC2046,SC2002,SC2034
+set -euo pipefail
+
+#
+# GovCMS disallowed modules check.
+#
+# This will lint exported configuration files to identify if
+# any disallowed module has been enabled via exports.
+#
+
+GOVCMS_CONFIG_FOLDER_PATH=${GOVCMS_CONFIG_FOLDER_PATH:-"config"}
+GOVCMS_CORE_EXTENTION_FILE_NAME=${GOVCMS_CORE_EXTENTION_FILE_NAME:-"core.extension.yml"}
+GOVCMS_DISALLOWED_MODULES=${GOVCMS_DISALLOWED_MODULES:-}
+
+config_file=$(find "$GOVCMS_CONFIG_FOLDER_PATH" -type f \( -name "$GOVCMS_CORE_EXTENTION_FILE_NAME" \))
+
+if [ -z "${config_file}" ]; then
+  echo "Coudn't find core.extension.yml file."
+  exit 0
+fi
+
+# Find the module name in a list (passed as a string).
+function is_module_enabled() {
+  echo "$2" | grep -c -E "^$1:?\s*\d"
+}
+
+FAILURES=""
+
+for file in $config_file; do
+  # Extract the list of enabled modules from core.extenions.yml:
+  ## the first sed command captures everything between `module:` and `theme:` keys in the file.
+  ## the returned result inclues `module:` and `theme:` so the second command removes the first and the last lines.
+  ## the final command removes tab indents.
+  enabled_modules=$(sed -n '/module:/,/theme:/p' "$file" | sed '1d;$d' | sed -e 's/[ \t]*//')
```

Comment:
Awesome `sed` wondering if `yq` might simplfy this a little, the govcms-ci image has `yq` available.

```
yq r "$file" 'module'
```
That should generate a list like:
```
admin_toolbar: 0
ctools_block: 0
datetime: 0
datetime_range: 0
dblog: 0
ga_login: 0
views_ui: 0
webform: 0
webform_ui: 0
workflows: 0
ds: 1
menu_link_content: 1
pathauto: 1
ds_extras: 2
views: 10
paragraphs: 11
linked_field: 100
update: 0
govcms: 1000
```

Reply:
We might be able to do 

```
yq r "$file" "module.$module_name"
```

Which will output the value or null eg.

```


That will output the value or null

```
module_name=paragraphs

yq r "$file" "module.$module_name"
11
```

Reply:
Awesome tip! I saw jq being used to parse json files but didn't notice yq was there as well. It significantly simplified the code.